packages/flutter/lib/src/gestures/monodrag.dart

```diff
@@ -515,6 +516,54 @@ class HorizontalDragGestureRecognizer extends DragGestureRecognizer {
   String get debugDescription => 'horizontal drag';
 }
 
+// A horizontal drag gesture detector that only responds to events on the
+// scrollbar's thumb and ignores everything else.
+// TODO(justinmc): I wanted to put this in cuperinto/scrollbar.dart along with
+// the long press detector, but it seems I can't override _isFlingGesture
+// outside of this file because it's private. Any better way to organize this?
+class ThumbHorizontalDragGestureRecognizer extends HorizontalDragGestureRecognizer {
+  ThumbHorizontalDragGestureRecognizer({
+    PointerDeviceKind kind,
+    Object debugOwner,
+    GlobalKey customPaintKey,
+  }) :  _customPaintKey = customPaintKey,
+        super(
+          kind: kind,
+          debugOwner: debugOwner,
+        );
+
+  final GlobalKey _customPaintKey;
+
+  // TODO(justinmc): Could I use some OOP magic to avoid duplicating this code
```

Comment:
If this code is on cupertino/scrollbar.dart then you could just have a file-private function.